master/buildbot/util/git.py

```diff
@@ -157,14 +157,18 @@ def _getSshDataPath(self):
         # ending with a slash
         path_module = self.build.path_module
 
-        workdir = self._getSshDataWorkDir().rstrip('/\\')
-        if path_module.isabs(workdir):
-            parent_path = path_module.dirname(workdir)
+        if self.build.hasProperty("builddir"):
```

Comment:
Would using builder name as part of the path solve the issue? That would make the solution not depend on whether `builddir` is set or not.

Reply:
Yes I agree, I just wasn't sure this information was available here, but I will change it